### DISCUSSION TITLE
Why does each component require file storage (as well as object storage)

### DISCUSSION THREAD

**Discussion by WoodyWoodsta** [grafana/mimir#5808] [Help and support] [answered]
I'm setting up a mimir deployment in kubernetes, and I'm not sure where to look to find out why each of the components that require a persistent volume requires that persistence in addition to object storage.

I'm trying to make a decision as to what storage driver to use for PVs and so I'd like to understand better why each components needs the persistence, or whether I can opt for `emptydir` and not have persistence across restarts or failures.

Is the persistence for caches and if so, what is the impact on the deployment if the cache is cleared?

**Comment by pstibrany** [comment, marked as answer]
Ingesters need persistent storage. Other components use storage as cache only (store-gateway, compactor) and can survive loss of data (although in case of store-gateways, they will need to re-sync possibly lot of data on restarts).

Other components don't require local persistent storage.

For details please take a look at [architecture overview](https://grafana.com/docs/mimir/latest/get-started/about-grafana-mimir-architecture/) and [description of individual components](https://grafana.com/docs/mimir/latest/references/architecture/components/).

**Reply by WoodyWoodsta** [reply]
Thank you, the second link is what I was looking for!

**Reply by raweber42** [reply]
Ok, but what about [this recommendation by the Mimir Docs](https://grafana.com/docs/mimir/latest/references/architecture/components/store-gateway/#:~:text=To%20avoid%20the%20store%2Dgateway%20having%20to%20re%2Ddownload%20the%20index%20header%20during%20subsequent%20restarts%2C%20we%20recommend%20running%20the%20store%2Dgateway%20with%20a%20persistent%20disk.) @pstibrany ?

**Reply by deniszh** [reply]
@raweber42 : it's recommended but not required. Store gateway will download data from s3 and reconstruct indices - although it will take some time and resources.
